src/main/java/org/primefaces/extensions/renderer/CommandButtonSingleClickRenderer.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.renderer;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import org.primefaces.component.commandbutton.CommandButton;
+import org.primefaces.component.commandbutton.CommandButtonRenderer;
+
+/**
+ * {@link CommandButton} renderer disabling the button while action is processed.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ * @since 8.0
+ */
+public class CommandButtonSingleClickRenderer extends CommandButtonRenderer {
+
+    protected static final String[] CLASS_CONFIRM = {"ui-confirmdialog-yes", "ui-confirmdialog-no"};
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        CommandButton button = (CommandButton) component;
+        if (button.isRendered() && !button.isDisabled() && !isConfirmation(button)) {
+            String widgetVar = button.resolveWidgetVar(context);
+            button.setOnclick(prefix(button.getOnclick(), toggle(widgetVar, false)));
+            button.setOncomplete(prefix(button.getOncomplete(), toggle(widgetVar, true)));
+        }
+        super.encodeEnd(context, component);
+    }
+
+    protected boolean isConfirmation(final CommandButton button) {
+        return button.getStyleClass() != null
+               && Stream.of(CLASS_CONFIRM).anyMatch(s -> s.contains(button.getStyleClass()));
```

Comment:
I don't think the condition is right, I think you mean `button.getStyleClass().contains(s)`

Reply:
Else the confirm dialog buttons will be disabled on click. You are right about the condition being inverted. I'll change it to what you suggested.